src/planetary_system.jl

```diff
@@ -352,6 +352,340 @@ function generate_period_and_sizes_power_law(s::Star, sim_param::SimParam; num_p
+end
+
+function cdf_power_law(x::Float64, x0::Float64, x1::Float64, α::Float64)
+    @assert x0 <= x <= x1
+    if α != -1
+        return (x^(α+1) - x0^(α+1))/(x1^(α+1) - x0^(α+1))
+    else # if α == -1
+        return log(x/x0)/log(x1/x0)
+    end
+end
+
+function invert_cdf_power_law(y::Float64, x0::Float64, x1::Float64, α::Float64)
+    @assert x0 < x1
+    if α != -1
+        return (x0^(α+1) + (x1^(α+1) - x0^(α+1))*y)^(1/(α+1))
+    else # if α == -1
+        return x0*(x1/x0)^y
+    end
+end
+
+function draw_segmented_uniform(segments::Array{Tuple{Float64,Float64},1})
+    # Note: "segments" must be non-overlapping
+    n_segs = length(segments) # number of segments
+    if n_segs == 0
+        println("No segments given; returning NaN.")
+        return NaN
+    end
+    for (i,seg) in enumerate(segments)
+        @assert 0 <= seg[1] < seg[2] <= 1 # check that all segments are valid
+        if i != n_segs
+            @assert segments[i][2] < segments[i+1][1] # check to make sure there are no overlapping segments
+        end
+    end
+
+    seg_lengths = [seg[2]-seg[1] for seg in segments]
+    sum_lengths = sum(seg_lengths)
+
+    u_draw = rand()
+    for (i,seg) in enumerate(segments)
+        # Map the union of the segments uniformly to the domain [0,1]:
+        if sum(seg_lengths[1:i-1])/sum_lengths <= u_draw < sum(seg_lengths[1:i])/sum_lengths
+            return rand(Uniform(seg[1], seg[2]))
+        end
+    end
+end
+
+function compute_unstable_Hill_regions_periods_given_planets(P::AbstractVector{Float64}, mass::AbstractVector{Float64}, insert_pl_mass::Float64, star_mass::Float64, sim_param::SimParam; ecc::AbstractVector{Float64}=zeros(length(P)), verbose::Bool=false)
+    @assert length(P) == length(mass) == length(ecc)
+    min_num_mutual_hill_radii = get_real(sim_param, "num_mutual_hill_radii")
+    order = sortperm(P)
+
+    P_segments_unstable = Tuple{Float64,Float64}[]
+    for pl in 1:length(P)
+        a = semimajor_axis(P[order[pl]], star_mass)
+        mu = mass[order[pl]]/star_mass
+        hill_radius = calc_hill_sphere(a, mu)
+        a_lower, a_upper = a*(1. - ecc[order[pl]]) - hill_radius*min_num_mutual_hill_radii, a*(1. + ecc[order[pl]]) + hill_radius*min_num_mutual_hill_radii
+        a_lower = max(a_lower, 0.) # set a_lower=0 if negative
+        @assert 0 <= a_lower < a_upper
+        P_lower, P_upper = period_given_semimajor_axis(a_lower, insert_pl_mass+star_mass), period_given_semimajor_axis(a_upper, insert_pl_mass+star_mass)
+        if verbose
+            println("P_blocked: ", (P_lower, P_upper))
+        end
+        push!(P_segments_unstable, (P_lower, P_upper))
+    end
+    return P_segments_unstable
+end
+
+function compute_unstable_mutualHill_regions_periods_given_planets(P::AbstractVector{Float64}, mass::AbstractVector{Float64}, insert_pl_mass::Float64, star_mass::Float64, sim_param::SimParam; ecc::AbstractVector{Float64}=zeros(length(P)), insert_pl_ecc::Float64=0., verbose::Bool=false)
```

Comment:
What about writing a more general function that takes a boolean for whether to use the mutual hill radius?  
You could then have a long-named functions that call the general one to allow for setting the function via configuration files.

Reply:
Good idea, I've combined the two functions for computing the unstable regions given previous planets into one function with a boolean for whether or not to use the mutual Hill radius as you suggested.

Reply:
And also combined the two functions for drawing the (unscaled) period of the next planet given previous planets, and defined long-named functions setting either use_mutualHill=true or use_mutualHill=false

Reply:
By the way, after a lot of testing (both before and after I addressed all your points), there is still a small bug somewhere (probably in this function) that I cannot quite track down. Occasionally, an unscaled period would be drawn that puts the cluster just inside our stability limit (i.e., using our "test_stability" function would return false, and printing out the minimum separation would give ~7.99... whereas our stability limit is Delta_c = 8). This happens maybe 1 in ~100,000 draws...

Reply:
Hmm.  
Are the difference small enough that it seems likely to just be finite precission arithmetic?  
Or might it be a difference in how some small parameter like the mass of the planet or eccentricity is being treated in the two?

Reply:
My first thought was that I was providing the "wrong" masses when computing the period from the semi-major axis. Indeed, when I first wrote the function to compute the unstable regions using mutual Hill radii, I was converting `a_lower` and `a_upper` to `P_lower` and `P_upper` using the mass of the existing planet at `a` (since that is what seemed natural to do in the case of using just the Hill radii). BUT, the conversion of the unstable region in semi-major axis to period should instead use the mass of the planet we are trying to insert, which I've fixed earlier on.
As written, the function using just the Hill radii ignores the eccentricity of the planet we are trying to insert, but I have been testing the function using the mutual Hill radii.
The difference seems too large to be just due to finite precision (I am seeing differences greater than 0.01 below Delta_c = 8)...

Reply:
I agree, that's too much to be finite precission arithmetic.  
I notice in some places you have expressions like (min_num_mutual_hill_radii/2.).  
You might want to check if you have included/excluded factors of 2 self-consistently.